keyboards/ergodox_ez/led_i2c.c

```diff
@@ -23,29 +23,41 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #    include "ergodox_ez.h"
 
 void rgblight_call_driver(LED_TYPE *led, uint8_t led_num) {
-    i2c_init();
-    i2c_start(0x84, ERGODOX_EZ_I2C_TIMEOUT);
+    uint16_t length = 0;
     int i = 0;
+    int j = 0;
+#    ifdef RGBW
+    int bytes_per_led = 4;
+#    else
+    int bytes_per_led = 3;
+#    endif
 #    if defined(ERGODOX_LED_30)
     // prevent right-half code from trying to bitbang all 30
     // so with 30 LEDs, we count from 29 to 15 here, and the
     // other half does 0 to 14.
     uint8_t half_led_num = RGBLED_NUM / 2;
+    length = half_led_num * bytes_per_led;
+    uint8_t data[length];
```

Comment:
This might be up to 60 bytes in the RGBW case; hopefully we have that amount available on the stack.  But the core option for reordering the LEDs for rgblight (`RGBLIGHT_LED_MAP`) would eat even more stack (`LED_TYPE led0[RGBLED_NUM]`).

Reply:
I looked into seeing if this could be split up into chunks (sending perhaps one or a few bytes at a time) to reduce the number of bytes on the stack, but my findings were inconclusive.

It looks like ErgoDox EZ sends the I2C data to an ICE5LP FPGA which translates the I2C data to the RGB LED data. The I2C comms isn't using the register write method send data to addressable registers, otherwise it would be trivially easy to break it up into smaller pieces. The I2C transmit is instead  just sending a stream of bytes to the FPGA, and I don't know if the FPGA is writing them sequentially to a ring buffer internally, or if the FPGA assumes each I2C transmit restarts at the first RGB LED.

I don't have an ErgoDox EZ to test with, so it would be good if someone with an ErogDox EZ with RGB LEDs can check (1) if the number of bytes on the stack actually causes a problem and (2) if so, whether the bytes can be sent one at a time through sequential transmit messages. If not, the path forward in this situation may be to continue using the lower level functions to assemble the I2C message one byte at a time.

Reply:
A quick check shows that this seems to be okay.  (with both rgb matrix and rgb light enabled, even). 

Also, not sure if it helps, but somebody hacked together custom replacement parts for the rgblight strip:
https://github.com/nicolai86/ErgoDox-EZ/tree/master/src

Reply:
While the hacked-together replacement for the rgblight strip doesn't reveal what is going on in the iCE5LP FPGA, it at least provides the data point that _at least one_ implementation assumes each I2C transmit restarts at the first RGB LED, which tells me there no good way to break the bytes up into smaller chunks. Combined with your quick check showing that it seems to work as-is without breaking it up into smaller chunks, I'll go ahead and mark this as resolved. Thank you!